File: android_app/src/main/java/me/timeto/app/ui/EventsView.kt
package me.timeto.app.ui

import androidx.compose.animation.animateColorAsState
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import me.timeto.app.*
import me.timeto.shared.vm.EventsVm

private val menuTopPadding = 8.dp
private val menuBottomPadding = menuTopPadding.goldenRatioUp().goldenRatioUp()

@Composable
fun EventsView() {

    val (vm, state) = rememberVm { EventsVm() }

    VStack(
        modifier = Modifier
            .fillMaxSize(),
    ) {

        if (state.isCalendarOrList) {
            EventsCalendarView(
                modifier = Modifier
                    .weight(1f),
            )
        } else {
            EventsListView(
                modifier = Modifier
                    .weight(1f),
            )
        }

        DividerBg(
            modifier = Modifier
                .padding(start = H_PADDING, end = TasksView__PADDING_END),
        )

        HStack(
            modifier = Modifier
                .padding(top = menuTopPadding, bottom = menuBottomPadding),
        ) {

            ModeButton(
                text = "Calendar",
                modifier = Modifier
                    .padding(start = H_PADDING - halfDpCeil),
                isActive = state.isCalendarOrList,
                onClick = {
                    vm.setIsCalendarOrList(true)
                },
            )

            ModeButton(
                text = "List",
                modifier = Modifier
                    .padding(start = 8.dp),
                isActive = !state.isCalendarOrList,
                onClick = {
                    vm.setIsCalendarOrList(false)
                },
            )
        }
    }
}

//
// Mode Button

@Composable
private fun ModeButton(
    text: String,
    modifier: Modifier,
    isActive: Boolean,
    onClick: () -> Unit,
) {

    Text(
        text = text,
        modifier = modifier
            .clip(shape = squircleShape)
            .clickable {
                onClick()
            }
            .background(animateColorAsState(if (isActive) c.blue else c.transparent).value)
            .padding(horizontal = 7.dp)
            .padding(top = 2.dp + onePx, bottom = 2.dp),
        color = if (isActive) c.white else c.text,
        fontSize = 14.sp,
        fontWeight = FontWeight.Medium,
    )
}

File: ios_app/iphone/EventsView.swift
import SwiftUI
import shared

private let menuTopPadding = 6.0
private let menuBottomPadding = menuTopPadding.goldenRatioUp().goldenRatioUp()

struct EventsView: View {

    @State private var vm = EventsVm()

    var body: some View {

        VMView(vm: vm) { state in

            VStack {

                if (state.isCalendarOrList) {
                    EventsCalendarView()
                } else {
                    EventsListView()
                }

                DividerBg()
                        .padding(.leading, H_PADDING)
                        .padding(.trailing, H_PADDING)

                HStack {

                    ModeButton(
                        text: "Calendar",
                        isActive: state.isCalendarOrList,
                        onClick: {
                            vm.setIsCalendarOrList(isCalendarOrList: true)
                        }
                    )
                            .padding(.leading, H_PADDING - halfDpCeil)

                    ModeButton(
                        text: "List",
                        isActive: !state.isCalendarOrList,
                        onClick: {
                            vm.setIsCalendarOrList(isCalendarOrList: false)
                        }
                    )
                            .padding(.leading, 8)

                    Spacer()
                }
                        .padding(.top, menuTopPadding)
                        .padding(.bottom, menuBottomPadding)
            }
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
        }
    }
}

//
// Mode Button

private let modeButtonShape = RoundedRectangle(cornerRadius: 8, style: .continuous)

private struct ModeButton: View {

    let text: String
    let isActive: Bool
    let onClick: () -> Void

    var body: some View {

        Button(
            action: {
                onClick()
            },
            label: {
                Text(text)
                        .font(.system(size: 16, weight: .medium))
                        .foregroundColor(isActive ? c.white : c.text)
                        .padding(.horizontal, 7)
                        .padding(.top, 3)
                        .padding(.bottom, 3)

            }
        )
                .background(modeButtonShape.fill(isActive ? c.blue : c.transparent))
    }
}
